harness/determined/common/experimental/experiment.py

```diff
@@ -166,6 +168,34 @@ def delete(self) -> None:
         """
         bindings.delete_DeleteExperiment(self._session, experimentId=self._id)
 
+    def download_code(self, output_dir: Optional[str] = None) -> str:
```

Comment:
What do you think about instead `download_model_def`? Or maybe "code" is better, I dunno. It's just ambiguous to this reader, and elsewhere the payload is "model def".

Reply:
Also aligns with CLI

Reply:
Further in that vein, how much work would it be to make the CLI use this method?

Reply:
yeah i was a bit torn on this myself. the CLI calls it `download_model_def`, but the web UI calls it "download experiment code". to me the web UI name makes more sense here, since "model def" makes me think it's just a single `model_def.py` file or something.

but i can also see an argument for making this consistent with the CLI (even though i don't like the CLI/REST API's naming here). 

i agree it's ambiguous to the reader. i didn't want to call it `download_experiment_code` since this was already a method on the Experiment resource, but that might be clearer? i dunno, what do you think?

Reply:
I like your choice better than the CLI's choice. And I think it makes sense to localize it from the web UI's. I guess I'm glad this is your PR rather than mine so you're the one who has to make the decision between better and more consistent.